src/main/java/de/tum/in/www1/artemis/repository/SolutionProgrammingExerciseParticipationRepository.java

```diff
@@ -14,5 +15,6 @@
 @Repository
 public interface SolutionProgrammingExerciseParticipationRepository extends JpaRepository<SolutionProgrammingExerciseParticipation, Long> {
 
+    @EntityGraph(attributePaths = "results")
```

Comment:
I think we should add "WithResults" to the method name

Reply:
This is not supported by the query syntax as far as I know.
I agree though that it should be part of the method name and therefore made it an explicit query with the `@Query` annotation.